3rdparty/indi-astromechfoc/astromech_focuser.cpp

```diff
@@ -0,0 +1,314 @@
+/*******************************************************************************
+ Copyright(c) 2019 Christian Liska. All rights reserved.
+
+ Implementation based on Lacerta MFOC driver
+ (written 2018 by Franck Le Rhun and Christian Liska).
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Library General Public
+ License version 2 as published by the Free Software Foundation.
+ .
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Library General Public License for more details.
+ .
+ You should have received a copy of the GNU Library General Public License
+ along with this library; see the file COPYING.LIB.  If not, write to
+ the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ Boston, MA 02110-1301, USA.
+*******************************************************************************/
+
+#include "astromech_focuser.h"
+#include "indicom.h"
+#include "connectionplugins/connectionserial.h"
+
+#include <cmath>
+#include <memory>
+#include <cstring>
+#include <unistd.h>
+
+// We declare an auto pointer to astromechanics_foc.
+static std::unique_ptr<astromechanics_foc> Astromechanics_foc(new astromechanics_foc());
+
+// Delay for receiving messages
+#define FOCUS_TIMEOUT  1000
+#define FOC_POSMAX_HARDWARE 9999
+#define FOC_POSMIN_HARDWARE 0
+
+void ISPoll(void *p);
+
+void ISGetProperties(const char *dev)
+{
+    Astromechanics_foc->ISGetProperties(dev);
+}
+
+void ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    Astromechanics_foc->ISNewSwitch(dev, name, states, names, n);
+}
+
+void ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n)
+{
+    Astromechanics_foc->ISNewText(dev, name, texts, names, n);
+}
+
+void ISNewNumber(const char *dev, const char *name, double values[], char *names[], int n)
+{
+    Astromechanics_foc->ISNewNumber(dev, name, values, names, n);
+}
+
+void ISNewBLOB(const char *dev, const char *name, int sizes[], int blobsizes[], char *blobs[], char *formats[],
+               char *names[], int n)
+{
+    INDI_UNUSED(dev);
+    INDI_UNUSED(name);
+    INDI_UNUSED(sizes);
+    INDI_UNUSED(blobsizes);
+    INDI_UNUSED(blobs);
+    INDI_UNUSED(formats);
+    INDI_UNUSED(names);
+    INDI_UNUSED(n);
+}
+
+bool astromechanics_foc::Disconnect() {
+    SetApperture(0);
+    MoveAbsFocuser(0);
+
+    return true;
+}
+
+void ISSnoopDevice(XMLEle *root)
+{
+    Astromechanics_foc->ISSnoopDevice(root);
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+astromechanics_foc::astromechanics_foc()
+{
+    setVersion(0, 1);
```

Comment:
version should be from the CMakeLists.txt file, see other drivers (INDI_ASTROMECHFOC_VERSION_MAJOR)